File: cases/taint/java/computed-property-positive/ComputedPropertyPositive.java
package dataflowbench.taint;

import java.lang.reflect.Field;

final class ComputedPropertyPositive {
    static final class Holder {
        int tainted;
        int clean;
    }

    static int dfb_source() { // DFB-SOURCE: computed-property-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: computed-property-sink

    static void run() throws Exception {
        Holder holder = new Holder();
        String key = "tainted";
        Field member = Holder.class.getDeclaredField(key); // DFB-WITNESS: computed-property-member
        member.setInt(holder, dfb_source()); // DFB-WITNESS: computed-property-store
        dfb_sink(member.getInt(holder));
    }
}

File: cases/taint/kotlin/computed-property-positive/ComputedPropertyPositive.kt
package dataflowbench

object ComputedPropertyPositive {
    class Holder {
        @JvmField var alpha: String = "clean"
        @JvmField var beta: String = "clean"
    }

    fun dfb_source(): String { // DFB-SOURCE: computed-property-input
        return "tainted"
    }

    fun dfb_sink(value: String) {} // DFB-SINK: computed-property-sink

    fun run() {
        val holder = Holder()
        val key = "alpha"
        val field = Holder::class.java.getDeclaredField(key)
        field.set(holder, dfb_source()) // DFB-WITNESS: computed-property-store
        dfb_sink(field.get(holder) as String)
    }
}
